proto/kv/v1/kv.proto

```diff
@@ -38,7 +44,7 @@ message KeyValuePutRequest {
   string collection = 1;
 
   // The unique key of the keyValue to put
-  string key = 2;
+  google.protobuf.Struct key = 2;
```

Comment:
Everything else looks good, doing some more thinking about this. Is struct too much for the keys here?

Will we be supporting deeply nested objects as keys, or can we flatten these into a simpler map/dict implementation?